atomic_reactor/plugins/post_pulp_sync.py

```diff
@@ -154,14 +134,44 @@ def set_auth(self, pulp):
             # Tell dockpulp
             pulp.set_certs(cer, key)
 
+    def get_dockercfg_credentials(self, docker_registry):
+        """
+        Read the .dockercfg file and return an empty dict, or else a dict
+        with keys 'basic_auth_username' and 'basic_auth_password'.
+        """
+        if not self.registry_secret_path:
+            return {}
+
+        json_secret_path = os.path.join(self.registry_secret_path,
+                                        '.dockercfg')
+        try:
+            with open(json_secret_path) as fp:
+                json_secret = json.load(fp)
+        except Exception:
+            msg = "failed to read registry secret"
+            self.log.error(msg, exc_info=True)
+            raise RuntimeError(msg)
+
+        try:
+            registry_creds = json_secret[docker_registry]
+        except KeyError:
+            self.log.warn('%s not found in .dockercfg',
+                          docker_registry)
+            return {}
+
+        return {
+            'basic_auth_username': registry_creds['username'],
```

Comment:
Not sure if email is required for some registries, but just to be safe we should fetch it as well